client/network_status.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/server"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"time"
+)
+
+func (cli *VanClient) NetworkStatus() ([]*types.SiteInfo, error) {
+
+	var sites *[]types.SiteInfo
+	var err error
+	err = utils.Retry(5*time.Second, 5, func() (bool, error) {
+		sites, err = server.GetSiteInfo(cli.Namespace, cli.KubeClient, cli.RestConfig)
+		if err != nil {
+			return false, err
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	versionCheckedSites := cli.checkSiteVersion(sites)
+
+	var services *[]types.ServiceInfo
+	err = utils.Retry(5*time.Second, 5, func() (bool, error) {
+		services, err = server.GetServiceInfo(cli.Namespace, cli.KubeClient, cli.RestConfig)
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var listSites []*types.SiteInfo
+
+	for _, site := range versionCheckedSites {
+
+		if len(site.Namespace) == 0 {
+			return nil, errors.New("unable to provide site information")
+		}
+
+		listLinks, err := cli.getSiteLinksStatus(site.Namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		listServicesAndTargets, err := cli.getServicesAndTargetsBySiteId(services, site.SiteId)
+		if err != nil {
+			return nil, err
+		}
+
+		newSite := types.SiteInfo{Name: site.Name, Namespace: site.Namespace, SiteId: site.SiteId, Url: site.Url, Version: site.Version, MinimumVersion: site.MinimumVersion, Links: listLinks, Services: listServicesAndTargets}
+
+		listSites = append(listSites, &newSite)
+
+	}
+	return listSites, nil
+}
+
+func (cli *VanClient) getSiteLinksStatus(namespace string) ([]string, error) {
+	lightRed := "\033[1;31m"
+	resetColor := "\033[0m"
+	var listLinks []string
+
+	if len(namespace) == 0 {
+		return nil, errors.New("Unspecified namespace")
+	}
+
+	mapLinkStatus, err := cli.getLinkStatusByNamespace(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	links := make([]string, 0, len(mapLinkStatus))
+	for connection := range mapLinkStatus {
+		links = append(links, connection)
+	}
+
+	for _, link := range links {
+		var formattedLink string
+
+		if mapLinkStatus[link].Connected {
+			formattedLink = link
+		} else {
+			formattedLink = fmt.Sprintf("%s%s(link not active)%s", lightRed, link, resetColor)
```

Comment:
What do you think about adding a blank space between the site id and the parenthesis.

Reply:
Agree on that, I will work on it